src/map/unit.cpp

```diff
@@ -1815,14 +1815,16 @@ TIMER_FUNC(unit_resume_running){
 
 /**
  * Sets the delays that prevent attacks and skill usage considering the bl type
- * Officially it just remembers the last attack time here and applies the delays during the comparison
+ * Officially for players it just remembers the last attack time here and applies the delays during the comparison
  * But we pre-calculate the delays instead and store them in attackabletime and canact_tick
- * Currently these delays are only applied to PCs
- * TODO: This function should also be called on attacks and cast begin
+ * For ground skills this also applies to mercenaries
+ * For non-PCs this function also handles setting of attack delay (which we also store in attackabletime)
+ * TODO: Currently this function is only called for normal attacks and parry events and is a work-in-progress
  * @param bl Object to apply attack delay to
  * @param tick Current tick
```

Comment:
If it is always current tick, why do we even make the caller send it?

Reply:
Using tick from the timer passed through the functions is common to everything we do in rAthena.